pkgs/unified_analytics/lib/src/utils.dart

```diff
@@ -81,6 +83,68 @@ Directory getHomeDirectory(FileSystem fs) {
   return fs.directory(home!);
 }
 
+/// Returns `true` if user has opted out of legacy analytics in Dart or Flutter
+///
+/// Checks legacy opt-out status for the Flutter
+/// and Dart in the following locations
+///
+/// Dart: `$HOME/.dart/dartdev.json`
+///
+/// Flutter: `$HOME/.flutter`
+bool legacyOptOut({
+  required FileSystem fs,
+  required Directory home,
+}) {
+  final File dartLegacyConfigFile =
+      fs.file(p.join(home.path, '.dart', 'dartdev.json'));
+  final File flutterLegacyConfigFile = fs.file(p.join(home.path, '.flutter'));
+
+  // Example of what the file looks like for dart
+  //
+  // {
+  //   "firstRun": false,
+  //   "enabled": false,  <-- THIS USER HAS OPTED OUT
+  //   "disclosureShown": true,
+  //   "clientId": "52710e60-7c70-4335-b3a4-9d922630f12a"
+  // }
+  try {
+    if (dartLegacyConfigFile.existsSync()) {
+      // Read in the json object into a Map and check for
+      // the enabled key being set to false; this means the user
+      // has opted out of analytics for dart
+      final Map<String, Object?> dartObj =
+          jsonDecode(dartLegacyConfigFile.readAsStringSync());
+      if (dartObj.containsKey('enabled') && dartObj['enabled'] == false) {
+        return true;
+      }
+    }
+  } catch (e) {
+    // Continue if the file was something we couldn't parse
+  }
+
+  // Example of what the file looks like for flutter
+  //
+  // {
+  //   "firstRun": false,
+  //   "clientId": "4c3a3d1e-e545-47e7-b4f8-10129f6ab169",
+  //   "enabled": false  <-- THIS USER HAS OPTED OUT
+  // }
+  try {
+    if (flutterLegacyConfigFile.existsSync()) {
+      // Same process as above for dart
+      final Map<String, Object?> flutterObj =
+          jsonDecode(dartLegacyConfigFile.readAsStringSync());
+      if (flutterObj.containsKey('enabled') && flutterObj['enabled'] == false) {
+        return true;
+      }
+    }
+  } catch (e) {
+    // Continue if the file was something we couldn't parse
+  }
```

Comment:
It would be safer, imo, to assume that if the file exists but can't be read then it probably exists because the user _did_ opt out. The error is likely to be rare, but there's less cost, in terms of user trust, if we don't collect analytics when they've opted in than if we do collect analytics when they've opted out.

Reply:
That's a thought, and agreed that it would be best in terms of user trust. I'll make the update for your review